File: C/Program19.c
/////////////////////////////////////////////////////////////////////////
//                                                                     //
//  File name :     Program19.c                                        //
//  Description :   Used to Check Perfect Number                       //
//  Author :        Pallavi Omprakash Malewar                          //
//  Date :          19/07/2025                                         //
//                                                                     //
/////////////////////////////////////////////////////////////////////////


#include<stdio.h>
#include<stdbool.h>

bool CheckPerfectNumber(int iNumber)
{
    int iCount = 0 ;
    int iSum = 0 ;

    if(iNumber < 0)             // Updator
    {
        iNumber = -iNumber ;
    }

    for( iCount = 1 ; iCount <= (iNumber / 2) ; iCount++)
    {
        if((iNumber % iCount) == 0 )
        {
            iSum = iSum + iCount ;
        }
    }

    if(iSum == iNumber)
    {
        return true ;
    }
    else 
    {
        return false ;
    }
}


int main()
{
    int iValue = 0 ;
    bool bRet = false ;

    printf("Enter Number : ") ;
    scanf("%d" , &iValue) ;

    bRet = CheckPerfectNumber(iValue) ;

    if(bRet == true)
    {
        printf("%d is a Perfect Number.\n" , iValue) ;
    }    
    else
    {
        printf("%d is not a Perfect Number.\n" , iValue);
    }   

    return 0;

}

// Time Complexity O(N/2)

File: JAVA/Program19.java
//////////////////////////////////////////////////////////////////////////////////////////
//                                                                                      //
//  File name :     Program19.java                                                      //
//  Description :   Used to Print Given Pattern                                         //
//  Author :        Pallavi Omprakash Malewar                                           //
//  Date :          25/08/2025                                                          //
//                                                                                      //
//////////////////////////////////////////////////////////////////////////////////////////


// Input : 4
// Output : *   *   *   *

/*
    Start
        Accept the frequency
        Iterate from 1 to that frequency
            Display * on screen
        continue
    Stop
*/


import java.util.Scanner;

class Pattern
{
    public void Display(int iNo)
    {
        int iCount = 0;

        if(iNo < 0)
        {
            iNo = -iNo;     // Filter
        }

        for(iCount = 1 ; iCount <= iNo ; iCount++)
        {
            System.out.print("*\t");
        }
        System.out.println("\n");
    }
}

class Program19
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.print("Enter Number : ");
        iValue = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.Display(iValue);

    }
}

// Time Complexity O(N)
// N is a Natural Number
// Where N >= 0 & N <= Infinity